test/e2erp/realrp_test.go

```diff
@@ -3,15 +3,84 @@
 package e2erp
 
 import (
+	"context"
+	"os"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/log"
 )
 
 var _ = Describe("Resource provider e2e tests [Real]", func() {
 	defer GinkgoRecover()
 
-	It("dummy real test", func() {
-		var err error
+	It("should not be possible for customer to mutate an osa scale set", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+		logger := logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": c.resourceGroup})
+
+		appRg := ApplicationResourceGroup(c.resourceGroup, c.resourceGroup, c.location)
+		Expect(appRg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("application resource group is %s", appRg)
+
+		// update the logger to use the application resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": appRg})
+
+		managedRg, err := ManagedResourceGroup(ctx, appRg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(appRg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("managed resource group is %s", managedRg)
+
+		// update the logger to use the managed resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": managedRg})
+
+		scaleSets, err := ScaleSets(ctx, logger, c.ssc, managedRg)
 		Expect(err).NotTo(HaveOccurred())
+		Expect(scaleSets).NotTo(And(BeNil(), BeEmpty()))
+		Expect(len(scaleSets)).Should(BeNumerically(">=", 3))
+
+		// TODO: get detailed error and match on them since we expect the customer to see errors with Code=ScopeLocked
+		var errs []error
+
+		logger.Info("Updating the scale set instance count")
+		errs = UpdateScaleSetsCapacityErrors(ctx, logger, c.ssc, c.ssvmc, managedRg)
+		Expect(errs).NotTo(BeNil())
+		Expect(len(errs)).To(BeEquivalentTo(len(scaleSets)))
+
+		logger.Info("Updating the scale set instance type")
+		errs = UpdateScaleSetsInstanceTypeErrors(ctx, logger, c.ssc, managedRg)
+		Expect(errs).NotTo(BeNil())
+		Expect(len(errs)).To(BeEquivalentTo(len(scaleSets)))
+
+		logger.Info("Updating the scale set SSH key")
```

Comment:
I would put each test in its own spec so we will not be maintaining a monolithic test but can be done in a follow-up.